src/networkd.c

```diff
@@ -30,6 +30,27 @@
 #include "util.h"
 
 
+/**
+ * wowlan_triggers= string for wpa_supplicant.conf
+ */
+char *
+wifi_wowlan_str(NetplanWifiWowlanFlag flag) {
+    if (flag & NETPLAN_WIFI_WOWLAN_TYPES[0].flag || flag >= NETPLAN_WIFI_WOWLAN_TCP) {
+        g_fprintf(stderr, "ERROR: unsupported wowlan_trigger mask: 0x%x\n", flag);
+        exit(1);
+    }
+    GString* str = NULL;
+    str = g_string_sized_new(200);
```

Comment:
hm, I'm a bit worried here, but where do we free up the memory allocated for this string? After g_string_sized_new() is used, the pointer then is lost before we can do g_string_free() on it.
Maybe we could make this function return a GString* and make it the responsibility of the caller to free the space? I wouldn't want us to leak memory even in such small amounts.

Reply:
I'll pass the existing GString* as an argument, and simply append to this. That way we don't need to allocate any extra memory at all.